src/win/fs-event.c

```diff
@@ -70,7 +70,7 @@ static int uv_relative_path(const WCHAR* filename,
   int filelen = wcslen(filename);
   if (dir[dirlen - 1] == '\\')
     dirlen--;
-  *relpath = uv__malloc((MAX_PATH + 1) * sizeof(WCHAR));
+  *relpath = uv__malloc((filelen - dirlen - 1 + 1) * sizeof(WCHAR));
```

Comment:
Sure, I did that to show the workings without reducing the reasoning, since the same calculation for `relpath` size is used a few lines later when copying. Would it be fine if I cache the `relpath` size calculation in an int `relpathlen`?